File: 10/03/test/DynamicLoadingTest.js
'use strict';
var webdriver = require('selenium-webdriver');
var test = require('selenium-webdriver/testing');
var assert = require('assert');
var DynamicLoadingPage = require('../pages/DynamicLoadingPage');

test.describe('Dynamic Loading', function() {
  this.timeout(30000);
  var driver;
  var dynamicLoading;

  test.beforeEach(function() {
    driver = new webdriver.Builder().forBrowser('firefox').build();
    dynamicLoading = new DynamicLoadingPage(driver);
  });

  test.afterEach(function() {
    driver.quit();
  });

  test.it('hidden element', function() {
    dynamicLoading.loadExample('1');
    dynamicLoading.finishTextPresent().then(function(elementDisplayed) {
      assert.equal(elementDisplayed, true, 'Finish text not displayed');
    });
  });

  test.it('rendered element', function() {
    dynamicLoading.loadExample('2');
    dynamicLoading.finishTextPresent().then(function(elementDisplayed) {
      assert.equal(elementDisplayed, true, 'Finish text not displayed');
    });
  });
});

File: 11/02/tests/dynamic_loading_test.py
import pytest
from pages import dynamic_loading_page


class TestDynamicLoading():

    @pytest.fixture
    def dynamic_loading(self, driver):
        return dynamic_loading_page.DynamicLoadingPage(driver)

    def test_hidden_element(self, dynamic_loading):
        dynamic_loading.load_example("1")
        assert(dynamic_loading.finish_text_present())

    def test_rendered_element(self, dynamic_loading):
        dynamic_loading.load_example("2")
        assert(dynamic_loading.finish_text_present())
